app/src/main/java/org/schabi/newpipe/info_list/holder/RepliesHandler.java

```diff
@@ -0,0 +1,192 @@
+import java.util.concurrent.Callable;
+
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
+import io.reactivex.rxjava3.annotations.NonNull;
+import io.reactivex.rxjava3.core.Single;
+import io.reactivex.rxjava3.core.SingleObserver;
+import io.reactivex.rxjava3.disposables.Disposable;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+
+public class RepliesHandler {
+    private final List<CommentsInfoItem> cachedReplies;
+    private final TextView showReplies;
+    private final RecyclerView repliesView;
+
+    public RepliesHandler(final TextView showReplies, final RecyclerView recyclerView) {
+        this.repliesView = recyclerView;
+        repliesView.setAdapter(new InfoListAdapter(repliesView.getContext()));
+        repliesView.setLayoutManager(new LinearLayoutManager(repliesView.getContext()));
+
+        this.showReplies = showReplies;
+        this.cachedReplies = new ArrayList<>();
+    }
+
+    static class GetMoreItemsCallable implements
+            Callable<ListExtractor.InfoItemsPage<CommentsInfoItem>> {
+        private CommentsInfo parentCommentInfo;
+        private CommentsInfoItem parentInfoItem;
+
+        public void setCallableParameters(
+                final CommentsInfo commentInfo, final CommentsInfoItem infoItem) {
+            parentCommentInfo = commentInfo;
+            parentInfoItem = infoItem;
+        }
+
+        @Override
+        public ListExtractor.InfoItemsPage<CommentsInfoItem> call() throws Exception {
+            return CommentsInfo.getMoreItems(parentCommentInfo, parentInfoItem.getReplies());
+        }
+    }
+
+    private Single<ListExtractor.InfoItemsPage<CommentsInfoItem>>
+    repliesSingle(final CommentsInfo parentCommentInfo,
+                      final CommentsInfoItem parentInfoItem) {
+
+        final GetMoreItemsCallable getMoreItems = new GetMoreItemsCallable();
+        getMoreItems.setCallableParameters(parentCommentInfo, parentInfoItem);
+        return Single.fromCallable(getMoreItems);
+
+    }
+
+    private SingleObserver<ListExtractor.InfoItemsPage<CommentsInfoItem>>
+    repliesObserver(final CommentsInfoItem parentInfoItem) {
+
+        return new SingleObserver<ListExtractor.InfoItemsPage<CommentsInfoItem>>() {
+
+            @SuppressLint("SetTextI18n")
+            @Override
+            public void onSubscribe(@NonNull final Disposable d) {
+                showReplies.setText("Setting up replies");
+            }
+
+            @Override
+            public void onSuccess(@NonNull final
+                                  ListExtractor.InfoItemsPage<CommentsInfoItem>
+                                          commentsInfoItemInfoItemsPage) {
+
+                final List<CommentsInfoItem> actualList
+                        = commentsInfoItemInfoItemsPage.getItems();
+
+                cachedReplies.addAll(actualList);
+                addRepliesToUI(parentInfoItem);
+            }
+
+            @SuppressLint("SetTextI18n")

```

Comment:
There is a reason why Android Studio gives you this warning: you shouldn't hardcode strings, put them in the `strings.xml` resource file instead.